EIPS/eip-1185.md

```diff
@@ -0,0 +1,80 @@
+
+## Specification
+
+The resolver profile to support DNS on ENS builds on the standalone resolver as defined in #137.
+
+Traditionally, DNS is a zone-based system in that all of the records for a zone are kept together in the same file.  This has the benefit of simplicity and atomicity of zone updates, but when transposed to ENS can result in significant gas costs for simple changes.  As a result, the resolver works on the basis of record sets.  A record set is uniquely defined by the tuple (domain, name, resource record type), for example the tuple (example.com, www.example.com, A) defines the record set of A records for the name www.example.com in the domain example.com.  A record set can contain 0 or more values, for example if www.example.com has A records 1.2.3.4 and 5.6.7.8 then the aforementioned tuple will have two values.
+
+The choice to work at the level of record sets rather than zones means that this specification cannot completely support some features of DNS, such as zone transfers and DNSSEC.  It would be possible to build a different resolver profile that works at the zone level, however it would be very expensive to carry out updates and so is not considered further for this EIP.
+
+The DNS resolver interface consists of two functions to set DNS information and two functions to query DNS information.
+
+### setDNSRecords(bytes32 node, bytes data)
+
+`setDNSRecords()` sets, updates or clears 1 or more DNS records for a given node.  It has function signature `0x0af179d7`.
+
+The arguments for the function are as follows:
+  - node: the nodehash of the fully-qualified domain in ENS for which to set the records.  Node hashes are defined in #137
+  - data: 1 or more DNS records in DNS wire format.  Any record that is supplied without a value will be cleared
+
+### clearDNSZone(bytes32 node)
+
+`clearDNSZone()` removes all DNS records for the domain.  It has function signature `0xad5780af`.
+
+Although it is possible to clear records individually with `setDNSRecords()` as described above this requires the owner to know all of the records that have been set (as the resolver has no methods to iterate over the records for a given domain), and might require multiple transactions.  `clearDNSZone()` removes all zone information in a single operation.
+
+The arguments for the function is as follows:
+  - node: the nodehash of the fully-qualified domain in ENS for which to clear the records.  Node hashes are defined in #137
+
+### dnsRecords(bytes32 node, bytes32 name, uint16 resource) view returns (bytes)
+
+`dnsRecords()` obtains the DNS records for a given node, name and resource.  It has function signature `0x2461e851`.
+
+The arguments for the function are as follows:
+  - node: the nodehash of the fully-qualified domain in ENS for which to set the records.  Node hashes are defined in #137
+  - name: the `keccak256()` hash of the name of the record in DNS wire format.
```

Comment:
Wouldn't it make sense to make this a namehash as well, for consistency?

Reply:
It used to be a namehash; you asked me on gitter to change it to this :)

If I remember correctly the rationale was that this mirrors the information provided in `setDNSRecords()`, i.e. the name is in DNS wire format.  Having different formats for the name in getting and setting records was considered potentially confusing (and having different formats for the namehash and the DNS name was considered okay because it clearly delineates between the ENS name and the DNS name).

Reply:
Passing the name in, in DNS wire format makes sense to me. If you're going to hash it, though, using two different schemes seems odd.

Reply:
Hashing is purely there to have a constant size value for the parameter.  It could easily be passed in unhashed, but not sure that it buys us anything.